python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -20,19 +20,127 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 from __future__ import annotations
 
-__all__ = ("SourceObjectTableAnalysisConfig", "SourceObjectTableAnalysisTask")
+__all__ = (
+    "SourceObjectTableAnalysisConfig",
+    "SourceObjectTableAnalysisTask",
+    "ObjectEpochTableConfig",
+    "ObjectEpochTableTask",
+)
 
 import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
 import numpy as np
 import pandas as pd
 from astropy.table import vstack
-from lsst.pipe.base import NoWorkFound
+from lsst.drp.tasks.gbdesAstrometricFit import calculate_apparent_motion
 from lsst.pipe.base import connectionTypes as ct
 from smatch import Matcher
 
 from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
 
 
+class ObjectEpochTableConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "skymap"),
+    defaultTemplates={
+        # "inputName": "sourceTable_visit",
+        # "inputCoaddName": "deep",
+        # "associatedSourcesInputName": "isolated_star_sources",
+        # "outputName": "sourceObjectTable",
+    },
+):
+    refCat = ct.Input(
+        doc="Catalog of positions to use as reference.",
+        name="objectTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+        deferLoad=True,
+        deferGraphConstraint=True,
+    )
+
+    refCatEpochMap = ct.Input(
+        doc="Healsparse map of mean epoch of refCat.",
+        name="deepCoadd_epoch_map_mean",
+        storageClass="HealSparseMap",
+        dimensions=("skymap", "tract", "band"),
+        multiple=True,
+        deferLoad=True,
+    )
+
+    objectEpochs = ct.Output(
+        doc="Catalog of epochs for refCat objects.",
+        name="objectEpochTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+    )
+
+
+class ObjectEpochTableConfig(pipeBase.PipelineTaskConfig, pipelineConnections=ObjectEpochTableConnections):
+    refCatBands = pexConfig.ListField(
+        doc=("Bands in refCat to be combined with `refCat_selectors` to build refCat column names."),
+        dtype=str,
+        default=["u", "g", "r", "i", "z", "y"],
+    )
+
+
+class ObjectEpochTableTask(pipeBase.PipelineTask):
+    """Collect mean epochs for the observations that went into each object."""
```

Comment:
I'm concerned that we don't have `epoch` or `mean_epoch` or [radecMjdTai](https://sdm-schemas.lsst.io/apdb.html#DiaObject.radecMjdTai) (as is used in the AP DiaObject table) already defined in our object tables. That feels like an oversight, and not something we should be computing in an analysis tool.

Reply:
See the thread here: https://lsstc.slack.com/archives/C2JPMCF5X/p1721331152029129. It might go into the cell-based coadds metadata. I agree that the calculation would ideally be done elsewhere, and I didn't want to make this a separate task at all, or have there be an object epoch product. However, the middleware prevents this from being done in `SourceObjectTableAnalysisTask`.

Reply:
Might want to put a TODO in this class's docstring that we expect to remove it eventually, once we are storing the measured epoch internally?